### ISSUE TITLE
How to train Custom dataset?

### ISSUE THREAD

**Issue by Lizhinwafu** [Pointcept/Pointcept#8]

**Comment by Gofinge** [comment]
Compared with PTv2 released code, we add support to ModelNet40, but still did not support ShapeNet Part Segmentation since it is a special task and might make the framework not elegant. But our code still remains some necessary components for ShapeNet, please further follow the discussion here -> https://github.com/Gofinge/PointTransformerV2/issues/28.

**Comment by Gofinge** [comment]
It really depends on the size of your individual plant data, S3DIS is pretty dense. Maybe you can first reduce the batch size and make the code runnable in your machine, then tune the parameter. Also `SpUNet` is efficient in both speed and memory, you can adopt it as a baseline.

**Comment by Gofinge** [comment]
`Can this code output visual test results？`
Try the visualization tool `save_point_cloud` in `pointcept.utils.visualization`. Kindly remind that the color should be scaled into [0, 1] instead of [0, 255]. For normalized color in the transformer, just `(color + 1) / 2` to fit the input required of the function.

`I only have one GPU, do the following two lines of code still need to be run?`

There is no need to multi-arch build for a single GPU. The comment was for Slurm contains multi-type of GPUs.

**Comment by Gofinge** [comment]
It seems that you were trying to test a model on `Pointcpt`. Pointcept architecture is slightly different from PTv2 released code. Did you try to load my released weight with pointcept? If so, map the checkpoint layer name to the pointcept version. Or retraining a weight for ModelNet40.

**Comment by Gofinge** [comment]
Let me check the bug. I didn't test the tester for ModelNet40.

**Comment by Gofinge** [comment]
Hay, I found the reason. It was because we default assume enable DDP (GPU >= 2), consequence, the saved weight name will have the prefix `module`. Our testing process only applies a single GPU, and we remove the first 7 characters (`module.`), then load the weight. You only train the model with a single GPU and cause the debug. I have already fixed the problem.

The basic config for spunet modelnet40 is not good enough since I did not tune the parameter, the following issue might be helpful to achieve a SOTA performance in ModelNet40: https://github.com/Gofinge/PointTransformerV2/issues/9

**Comment by Gofinge** [comment]
` I also encountered this error during testing while training s3dis. I think that's also the reason above.`
Yes, same reason (train by a single GPU).

**Comment by Gofinge** [comment]
Using different sampling rates for different categories of points has the following two risks: 1. Introducing category information to the data (based on sparse classification); 2. Uneven density brings perception difficulties.

My suggestion is:
1. Try LovaszLoss;
2. Wait for our code for (CAC & Ceco). It should be May~June after the NeurIPS deadline.

**Comment by Carl-Yang98** [comment]
> Good work. -- I will retrain and test again. -- I also encountered this error during testing while training s3dis. model.load_state_dict(state_dict, strict=True) ## I can test by changing True to False, but I don't know what impact it will have on the test results. -- I think that's also the reason above. ![image](https://user-images.githubusercontent.com/33644565/230045008-23c09d3e-85c4-4e6f-9e9e-45532ec19b85.png) ![image](https://user-images.githubusercontent.com/33644565/230045093-b7ac2fb3-a764-4276-997f-fad9f7a4257d.png)

Hey，I had met the same error and I had solved it. 
change the strict=True to strict=False will effect the result.
Here is my solution: 
![图片](https://user-images.githubusercontent.com/54114937/231387243-b09ec232-4f88-45eb-ba8f-db4c106fc1bc.png)

**Comment by Gofinge** [comment]
For tensorboard:
```
 tensorboard --logdir $POINTCEPT_PATH/exp/s3dis/ --bind --port=$PORT
```

For log file:
```
vim exp/s3dis/$EXP_NAME/train.log
```

For images, save them on your local PC or modify tensorboard log module.

**Comment by Gofinge** [comment]
Please considering write an additional instance segmenter and adding a post process in the segmenter.
